Composer/packages/extensions/adaptive-flow/src/adaptive-flow-renderer/utils/expression/widgetExpressionEvaluator.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { Templates } from 'botbuilder-lg';
+import { ExpressionParser } from 'adaptive-expressions';
+
+// Capture patterns that are strictly '${EXPRESSION}'
+//   - '${action.condition}'
+//   - '${if(action.value, concat(action.value, "(Event)"))}'
+const ValueAccessPattern = new RegExp(/^\$\{([^}]+)\}$/);
+
+// Catpure patterns that include '...${...}...'
+const StringTemplatePattern = new RegExp(/\$\{.+\}/);
```

Comment:
Personally think it is not the best solution to achieve the template result.
Language generation is aimed to help people to generate the result based on the template and data, so, if there are some gaps between the requirements and the features provided by LG, please file an issue on SDK side, and the LG team would follow up.

Reply:
I understand it could (and indeed) be ambigous to mix LG and Expression together. The confusion comes from LG users always expects `${expression}` to return a string; while Expression users expects to write `expression` directly.

Actually inspired from our discussion last week, I do have a backup solution that using `=if(x, a, b)` to mark a field as an Expression. How do you think about that?

Reply:
After a offline discussion, we finally decided to use a pure expression plan, where `"=action.x"` is evaluated by the `ValueExpression` class.